common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -206,6 +209,24 @@ def handle_ajax(self, dispatch, data):  # TODO: bounds checking
                 self.position = 1
             return json.dumps({'success': True})
 
+        if dispatch == 'get_completion':
+            if not self.is_user_authenticated:
```

Comment:
This has to be a function call (presumably passing in `data` for the `context` param).

Reply:
Also, you may want to write a unit test to cover the case where the user is not authenticated.

Reply:
@iloveagent57 What do you mean by "This" in your first comment? Is the the code block underneath the if condition on dispatch?